### PR TITLE
[MXNET-593] Adding 2 tutorials on Learning Rate Schedules

### PR DESCRIPTION
**Pull request by thomelane**
## Description ##

First tutorial is on learning rate schedules in `mxnet.lr_scheduler`, how to use them with Optimizer and Trainer in Gluon and how to implement a custom learning rate schedule. Contains visualizations of the schedules.

Second tutorial shows how to use common learning rate schedules such as Cyclical schedules, SDG with Restarts, Warm-up and Cool-down. Many paper references are provided. Contains visualizations of the schedules.

Added to tutorial testing and index page.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: docs/tutorials/gluon/learning_rate_schedules.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+    plt.xlabel("Iteration")
+    plt.ylabel("Learning Rate")
+    plt.show()
+```
+
+## Schedules
+
+### Stepwise Decay Schedule
+
+One of the most commonly used learning rate schedules is called stepwise decay, where the learning rate is reduced by a factor at certain intervals. MXNet implements a `FactorScheduler` for equally spaced intervals, and `MultiFactorScheduler` for greater control. We start with an example of halving the learning rate every 250 iterations.
```
Comment (mklissa) [inline docs/tutorials/gluon/learning_rate_schedules.md]: "One of the most commonly used learning rate schedules" - I would say "schedule" instead of "schedules"

**Comment by mklissa** [comment]
Very informative and well-written! Good job!

File: docs/tutorials/gluon/learning_rate_schedules.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+```python
+class CosineAnnealingSchedule():
+    def __init__(self, min_lr, max_lr, cycle_length):
+        self.min_lr = min_lr
+        self.max_lr = max_lr
+        self.cycle_length = cycle_length
+        
+    def __call__(self, iteration):
+        if iteration <= self.cycle_length:
+            unit_cycle = (1 + math.cos(iteration*math.pi/self.cycle_length))/2
```
Comment (KellenSunderland) [inline docs/tutorials/gluon/learning_rate_schedules.md]: nit: could add spaces around * and / operators to be consistent (and pass some linters).

File: docs/tutorials/gluon/learning_rate_schedules.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+Epoch: 5; Batch 0; Loss 0.035965; LR 0.003000 <!--notebook-skip-line-->
+
+Epoch: 6; Batch 0; Loss 0.005169; LR 0.003000 <!--notebook-skip-line-->
+
+Epoch: 7; Batch 0; Loss 0.017087; LR 0.003000 <!--notebook-skip-line-->
+
+Epoch: 8; Batch 0; Loss 0.039928; LR 0.000300 <!--notebook-skip-line-->
+
+Epoch: 9; Batch 0; Loss 0.003349; LR 0.000300 <!--notebook-skip-line-->
```
Comment (KellenSunderland) [inline docs/tutorials/gluon/learning_rate_schedules.md]: Epoch: 9; Batch 0; Loss 0.003349; LR 0.000300 should be Epoch: 9; Batch 0; Loss 0.003349; LR 0.0000300
I think right?  (i.e. there's a missing 0 in LR).

File: docs/tutorials/gluon/learning_rate_schedules_advanced.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        cycle_length: iterations between start and finish (int)
+        inc_fraction: fraction of iterations spent in increasing stage (float)
+        """
+        self.min_lr = min_lr
+        self.max_lr = max_lr
+        self.cycle_length = cycle_length
+        self.inc_fraction = inc_fraction
+        
+    def __call__(self, iteration):
+        if iteration <= self.cycle_length*self.inc_fraction:
```
Comment (KellenSunderland) [inline docs/tutorials/gluon/learning_rate_schedules_advanced.md]: Nit: again I would be consistent and alway have a space in between operators *, /, etc.
Same comment for code below.

File: docs/tutorials/gluon/learning_rate_schedules_advanced.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+plot_schedule(schedule)
+```
+
+
+![png](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/lr_schedules/adv_cooldown.png) <!--notebook-skip-line-->
+
+
+#### 1-Cycle: for "Super-Convergence"
+
+Cool-down is used in the "1-Cycle" schedule proposed by [Leslie N. Smith, Nicholay Topin (2017)](https://arxiv.org/abs/1708.07120) to train neural networks very quickly in certain circumstances (coined "super-convergence"). We implement a single and symmetric cycle of the triangular schedule above (i.e. `inc_fraction=0.5`), followed by a cool-down period of `cooldown_length` iterations.
```
Comment (KellenSunderland) [inline docs/tutorials/gluon/learning_rate_schedules_advanced.md]: Super-convergence is also introduced in the previous section's description.  Seems a bit redundant here.

File: docs/tutorials/gluon/learning_rate_schedules.md [hunk 6]
```diff
@@ -0,0 +1,317 @@
+
+# Learning Rate Schedules
+
+Setting the learning rate for stochastic gradient descent (SGD) is crucially important when training neural network because it controls both the speed of convergence and the ultimate performance of the network. One of the simplest learning rate strategies is to have a fixed learning rate throughout the training process. Choosing a small learning rate allows the optimizer find good solutions but this comes at the expense of limiting the initial speed of convergence. Changing the learning rate over time can overcome this tradeoff.
```
[the remark this answers is not in the document]
Comment (thomelane) [reply docs/tutorials/gluon/learning_rate_schedules.md]: Corrected.

**Comment by thomelane** [comment]
@KellenSunderland thanks for the review and feedback! Made changes as suggested, and fixed off by one.

**Comment by thomelane** [comment]
@indhub believe this is good to merge, since made adjustments as per @KellenSunderland feedback.

**Comment by thomelane** [comment]
JIRA: https://issues.apache.org/jira/browse/MXNET-593
